contracts/HubPool.sol

```diff
@@ -270,19 +287,18 @@ contract HubPool is Testable, Lockable, MultiCaller, Ownable {
         // Decrement the unclaimedPoolRebalanceLeafCount.
         refundRequest.unclaimedPoolRebalanceLeafCount--;
 
-        // Transfer the bondAmount to back to the proposer, if this was not done before for this refund bundle.
-        if (!refundRequest.proposerBondRepaid) {
-            refundRequest.proposerBondRepaid = true;
+        // Transfer the bondAmount to back to the proposer, if this the last executed leaf. Only sending this once all
+        // leafs have been executed acts to force the data worker to execute all bundles or they wont receive their bond.
+        //TODO: consider if we want to reward the proposer. if so, this is where we should do it.
+        if (refundRequest.unclaimedPoolRebalanceLeafCount == 0)
             bondToken.safeTransfer(refundRequest.proposer, bondAmount);
-        }
 
-        // TODO call into canonical bridge to send PoolRebalance.netSendAmount for the associated
-        // PoolRebalance.tokenAddresses, to the target PoolRebalance.chainId. this will likely happen within a
-        // x_Messenger contract for each chain. these messengers will be registered in a separate process that will follow
-        // in a later PR.
+        _sendTokensToTargetChain(poolRebalance, refundRequestId);
+        _executeRelayerRefundOnTargetChain(poolRebalance, refundRequestId);
+
         // TODO: modify the associated utilized and pending reserves for each token sent.
 
-        emit RelayerRefundExecuted(relayerRefundRequestId, poolRebalance, msg.sender);
+        emit RelayerRefundExecuted(refundRequestId, poolRebalance, msg.sender);
```

Comment:
1. Can we enumerate the values in this struct so they're decodable on etherscan?
2. Do we need to emit all of this data? With the size of these structs and the number of times this method gets called, I could see these getting expensive. Do you think we can drop some or most of this struct leaving only the most important info?

Reply:
good point & I agree! I will enumerate this to the minimum required data.